packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -555,11 +579,20 @@ const NavBar = <AdditionalEnvironmentProperties extends {}>(
   const useFullRedirectsForLinks = Boolean(
     props.environment.useFullRedirectsForLinks
   );
+  const location = useLocation();
 
-  const menuVisibilities = useApplicationContext(
-    (context) => context.menuVisibilities
+  const projectPermissions: TProjectPermissions = React.useMemo(
+    () => ({
+      permissions: normalizeAllAppliedPermissions(props.project),
+      actionRights: normalizeAllAppliedActionRights(props.project),
+      dataFences: normalizeAllAppliedDataFences(props.project),
+    }),
+    [props.project]
+  );
+  const menuVisibilities = React.useMemo(
+    () => normalizeAllAppliedMenuVisibilities(props.project),
+    [props.project]
```

Comment:
Normalizing the data is usually done in the application context (connectors). We need to do it manually here since we're skipping the application context now.